src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesGeneratorTest.java

```diff
@@ -124,95 +125,97 @@ public void shouldHandleExtraWhitespace() {
-        assertThat(INCORRECT_ORDER, names.get(13), is(new PersonName("O", "Arthur")));
-        assertThat(INCORRECT_ORDER, names.get(14), is(new PersonName("Coates", "Arthur")));
-        assertThat(INCORRECT_ORDER, names.get(15), is(new PersonName("Brian", "O")));
-        assertThat(INCORRECT_ORDER, names.get(16), is(new PersonName("O", "Brian")));
-        assertThat(INCORRECT_ORDER, names.get(17), is(new PersonName("Coates", "Brian")));
-        assertThat(INCORRECT_ORDER, names.get(18), is(new PersonName("Coates", "O")));
+        assertThat(INCORRECT_ORDER, names.get(3), is(new PersonName("Brian", "O Coates")));
+        assertThat(INCORRECT_ORDER, names.get(4), is(new PersonName("ArthurBrian", "OCoates")));
+        assertThat(INCORRECT_ORDER, names.get(5), is(new PersonName("Arthur", "Coates")));
+        assertThat(INCORRECT_ORDER, names.get(6), is(new PersonName("Brian", "Coates")));
+        assertThat(INCORRECT_ORDER, names.get(7), is(new PersonName("O", "Coates")));
+        assertThat(INCORRECT_ORDER, names.get(8), is(new PersonName("Arthur", "Brian")));
+        assertThat(INCORRECT_ORDER, names.get(9), is(new PersonName("Arthur", "O")));
+        assertThat(INCORRECT_ORDER, names.get(10), is(new PersonName("Brian", "Arthur")));
+        assertThat(INCORRECT_ORDER, names.get(11), is(new PersonName("Coates", "Arthur")));
+        assertThat(INCORRECT_ORDER, names.get(12), is(new PersonName("Brian", "O")));
+        assertThat(INCORRECT_ORDER, names.get(13), is(new PersonName("O", "Brian")));
+        assertThat(INCORRECT_ORDER, names.get(14), is(new PersonName("Coates", "Brian")));
+        assertThat(INCORRECT_ORDER, names.get(15), is(new PersonName("Coates", "O")));
+
+        assertThat(DEDUPLICATION, names, not(is(new PersonName("Arthur", "O Coates"))));
+        assertThat(DEDUPLICATION, names, not(is(new PersonName("OCoates", "ArthurBrian"))));
+        assertThat(DEDUPLICATION, names, not(is(new PersonName("O", "Arthur"))));
     }
 
     @Test
     public void shouldUseFirstFourAndLastThreeNamesIfOverSevenProvided() {
         List<PersonName> candidateNames = NameMatchingCandidatesGenerator.generateCandidateNames("A B C D E", "F G H");
 
-        assertThat(candidateNames.size(), is(71));
-
-        candidateNames = candidateNames.stream().
-                filter(name -> !name.lastName().equals("F G H")).
-                collect(Collectors.toList());
-
-        assertThat(candidateNames.size(), is(66)); // Expect there to be 5 candidates of retained first names paired with the entire lastname "F G H"
+        assertThat(candidateNames, hasItems(
+                new PersonName("A B C D", "F G H"),
+                new PersonName("B", "F G H"),
+                new PersonName("C", "F G H"),
+                new PersonName("D", "F G H")
+        ));
+        assertThat(DEDUPLICATION, candidateNames, not(hasItem(new PersonName("A", "F G H"))));
```

Comment:
Should this be split into 2 tests?  It tests both >7 name and multiple small word surnames.

Reply:
I have split this into two more logical tests.